src/components/TimelineMarker/index.js

```diff
@@ -10,18 +13,97 @@ import StyledLabel from './styled/label';
 import StyledDatetime from './styled/datetime';
 import StyledDescription from './styled/description';
 import StyledBody from './styled/body';
+import ExpandCollapseButton from './expandCollapseButton';
+import removeItemFromArray from '../AccordionSection/removeItemFromArray';
+import isInArray from '../AccordionSection/isInArray';
 
 /**
- * The TimelineMarker displays one event of an item timeline.
+ * The TimelineMarker displays one event of an item's timeline. It's generally used to compose the ActivityTimeline component.
  * @category Layout
  */
 export default function TimelineMarker(props) {
-    const { icon, label, description, datetime, children, className, style } = props;
+    const {
+        icon,
+        name,
+        isLoading,
+        label,
+        description,
+        datetime,
+        children,
+        className,
+        style,
+    } = props;
+    const {
+        isVariantAccordion,
```

Comment:
I think we should pass variant, since if we add new ones in the future then is more easy to make things here based on the variant